### PR TITLE
Fix "Sizes of columns doesn't match" error with SimpleAggregateFunction

### PR DESCRIPTION
**Pull request by bgranvea**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry:
Fix the "Sizes of columns doesn't match" error that might appear when using aggregate function columns.

This fixes the "Sizes of columns doesn't match" error which can appear when columns using SimpleAggregateFunction are merged. This bug didn't show before, I think it appears when using TTL and maybe the ttl_only_drop_parts setting.

### CODE REVIEW

**Comment by filimonov** [comment]
Is it possible to create a test case?

**Comment by alexey-milovidov** [comment]
@filimonov Yes, it is possible an is required.

**Comment by bgranvea** [comment]
so far I haven't been able to find a simple test case. There must be some case where AggregatingSortedBlockInputStream.merge is called with empty blocks to merge. In that case,  we insert an extra value in columns with SimpleAggregateFunction types. The fix is easy.

I have seen this bug on my test platform with Clickhouse 19.14.7.15 after inserting data for more than 2 days (and table TTL=1 day). With the fix, no more errors.

**Comment by bgranvea** [comment]
I've reproduced the bug on my test database but still haven't a simple test case.

What happens is that 2 empty parts (0 rows) are merged in background. AggregatingSortedBlockInputStream wrongly generates 1 value for SimpleAggregateFunction columns (see my fix). When the result block is written, there is a check in Block::checkNumberOfRows which detects that columns don't have the same size.

Do you have an idea how to obtain 2 empty parts in a simple test case? If I insert old values in a table with TTL, all my parts are always consolidated to a single empty part.

Here is the stack trace:
```
2019.11.06 14:17:15.098525 [ 12 ] {} <Error> sdwanLocal.flow_data_10minute_mvstore: DB::StorageReplicatedMergeTree::queueTask()::<lambda(DB::StorageReplicatedMergeTree::LogEntryPtr&)>: Code: 9, e.displayText() = DB::Exception: Sizes of columns doesn't match: customer: 0, wan_access_bw_ingress_src: 1, Stack trace:

0. 0x55fdd596738e StackTrace::tryCapture() /root/ClickHouse/dbms/src/Common/StackTrace.cpp:227
1. 0x55fdd596722d StackTrace::StackTrace() /root/ClickHouse/dbms/src/Common/StackTrace.cpp:191
2. 0x55fdd5966784 DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int) /root/ClickHouse/dbms/src/Common/Exception.h:27
3. 0x55fdd9ddf952 DB::Block::checkNumberOfRows(bool) const /root/ClickHouse/dbms/src/Core/Block.cpp:248
4. 0x55fdda5312e8 DB::MergedBlockOutputStream::writeImpl(DB::Block const&, DB::PODArray<unsigned long, 4096ul, Allocator<false, false>, 15ul, 16ul> const*) /root/ClickHouse/dbms/src/Storages/MergeTree/MergedBlockOutputStream.cpp:238
5. 0x55fdda52fe84 DB::MergedBlockOutputStream::write(DB::Block const&) /root/ClickHouse/dbms/src/Storages/MergeTree/MergedBlockOutputStream.cpp:90
6. 0x55fdda4a0bed DB::MergeTreeDataMergerMutator::mergePartsToTemporaryPart(DB::FutureMergedMutatedPart const&, DB::MergeListEntry&, DB::TableStructureReadLockHolder&, long, DB::DiskSpaceMonitor::Reservation*, bool, bool) /root/ClickHouse/dbms/src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp:758
7. 0x55fdda3b420d DB::StorageReplicatedMergeTree::tryExecuteMerge(DB::ReplicatedMergeTreeLogEntry const&) /root/ClickHouse/dbms/src/Storages/StorageReplicatedMergeTree.cpp:1097
8. 0x55fdda3b25d3 DB::StorageReplicatedMergeTree::executeLogEntry(DB::ReplicatedMergeTreeLogEntry&) /root/ClickHouse/dbms/src/Storages/StorageReplicatedMergeTree.cpp:930
9. 0x55fdda3c2f8b DB::StorageReplicatedMergeTree::queueTask()::{lambda(std::shared_ptr<DB::ReplicatedMergeTreeLogEntry>&)#2}::operator()(std::shared_ptr<DB::ReplicatedMergeTreeLogEntry>&) const /root/ClickHouse/dbms/src/Storages/StorageReplicatedMergeTree.cpp:2201
10. 0x55fdda3e91c5 std::_Function_handler<bool (std::shared_ptr<DB::ReplicatedMergeTreeLogEntry>&), DB::StorageReplicatedMergeTree::queueTask()::{lambda(std::shared_ptr<DB::ReplicatedMergeTreeLogEntry>&)#2}>::_M_invoke(std::_Any_data const&, std::shared_ptr<DB::ReplicatedMergeTreeLogEntry>&) /usr/include/c++/9/bits/std_function.h:287
11. 0x55fdda56abf3 std::function<bool (std::shared_ptr<DB::ReplicatedMergeTreeLogEntry>&)>::operator()(std::shared_ptr<DB::ReplicatedMergeTreeLogEntry>&) const /usr/include/c++/9/bits/std_function.h:689
12. 0x55fdda560083 DB::ReplicatedMergeTreeQueue::processEntry(std::function<std::shared_ptr<zkutil::ZooKeeper> ()>, std::shared_ptr<DB::ReplicatedMergeTreeLogEntry>&, std::function<bool (std::shared_ptr<DB::ReplicatedMergeTreeLogEntry>&)>) /root/ClickHouse/dbms/src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp:1117
13. 0x55fdda3c3c60 DB::StorageReplicatedMergeTree::queueTask() /root/ClickHouse/dbms/src/Storages/StorageReplicatedMergeTree.cpp:2163
14. 0x55fdda3cb7bd DB::StorageReplicatedMergeTree::startup()::{lambda()#1}::operator()() const /root/ClickHouse/dbms/src/Storages/StorageReplicatedMergeTree.cpp:2912
15. 0x55fdda3e9544 std::_Function_handler<DB::BackgroundProcessingPoolTaskResult (), DB::StorageReplicatedMergeTree::startup()::{lambda()#1}>::_M_invoke(std::_Any_data const&) /usr/include/c++/9/bits/std_function.h:287
16. 0x55fdda4390f8 std::function<DB::BackgroundProcessingPoolTaskResult ()>::operator()() const /usr/include/c++/9/bits/std_function.h:689
17. 0x55fdda437940 DB::BackgroundProcessingPool::threadFunction() /root/ClickHouse/dbms/src/Storages/MergeTree/BackgroundProcessingPool.cpp:203
18. 0x55fdda4369a3 DB::BackgroundProcessingPool::BackgroundProcessingPool(int)::{lambda()#1}::operator()() const /root/ClickHouse/dbms/src/Storages/MergeTree/BackgroundProcessingPool.cpp:70

```

**Comment by bgranvea** [comment]
ok, I have a test case with ALTER ... DELETE instead of TTL but the result is the same. I add it to the pull request.

File: dbms/tests/queries/0_stateless/01047_simple_aggregate_sizes_of_columns_bug.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+CREATE TABLE column_size_bug (date_time DateTime, value SimpleAggregateFunction(sum,UInt64)) ENGINE = AggregatingMergeTree PARTITION BY toStartOfInterval(date_time, INTERVAL 1 DAY) ORDER BY (date_time);
+
+INSERT INTO column_size_bug VALUES(now(),1);
+INSERT INTO column_size_bug VALUES(now(),1);
+
+ALTER TABLE column_size_bug DELETE WHERE value=1;
+
+-- wait for DELETE
+SELECT sleep(1);
```
Comment (alexey-milovidov) [inline dbms/tests/queries/0_stateless/01047_simple_aggregate_sizes_of_columns_bug.sql]: That will make the test flacky (random hangs for more than one sec. is typical in our CI and using `sleep` is not viable at all). But don't worry, I'll fix it by myself.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**